### DISCUSSION TITLE
Inline stylesheets in initial HTML

### DISCUSSION THREAD

**Discussion by Timer** [vercel/next.js#62363] [Ideas]
Instead of referencing a stylesheet via a `<link>` tag, we could inline its content in the initial HTML. This should result in a faster FCP. We need to do extensive measurement/testing before we'd consider doing this by default.

As an added bonus, this will help fix some browser-specific bugs (like Chrome incorrectly rendering a transition, #20540).

**Comment by xlkang** [comment]
Is there a specific time line for this issue?

**Comment by samcx** [comment]
Moving this to <picture data-single-emoji=":nextjs:" title=":nextjs:"><img class="emoji" width="20" height="auto" src="https://emoji.slack-edge.com/T0CAQ00TU/nextjs/2a85d633ae594556.png" alt=":nextjs:" align="absmiddle"></picture> [discussions](https://github.com/vercel/next.js/discussions) since this is a feature request!
